### ISSUE TITLE
TextAlignment renders text in wrong location for External Fonts

### ISSUE THREAD

**Issue by Niaxor** [AvaloniaUI/Avalonia#17510]
### Describe the bug

Avalonia does not render open type fonts loaded as a static resource in the correct location in certain circumstances. The circumstance seems to depend on the text length, text alignment, and previewer zoom %.

For example, this center aligned text is rendering as right-aligned:

![image](https://github.com/user-attachments/assets/5eeca490-0315-4cc8-8c04-433f8941ca0e)

If I left-align it it appears as centered:

![image](https://github.com/user-attachments/assets/5d1c25b5-0053-4a78-90b9-db3aa0c36f25)

If I removed a single 't' from the end of the string it then renders in the correct location....?

I have tested with two fonts, both open types as static resource, both behaved inconsistently. 
Using a regular system font like Arial works as intended. 

### To Reproduce

Ive uploaded a repro project -
[fonttest.zip](https://github.com/user-attachments/files/17772023/fonttest.zip)

1. Build and run, and the text will appear right-aligned, when it should be centre-aligned.
2. By modifying the text length, the text placement will vary.
3. You may also use the previewer, and modifying the zoom percentage also seems to trigger this issue, depending on the zoom percentage.

### Expected behavior

Open type fonts as a static resource are rendered in their expected location (based on layout settings). 

### Avalonia version

11.2

### OS

Windows

### Additional context

Rider 2024.1.4

**Comment by Niaxor** [comment]
Wrong repo to post this.

**Comment by Niaxor** [comment]
I have modified my description of this issue and uploaded a repro project. I believed this was an issue with the Rider previewer but am now aware it happens at runtime too, so this must be an issue with Avalonia itself. 

I do not yet have a workaround, this is rather critical because the basic building blocks of the UI is simply not rendering inconsistently.

**Comment by Niaxor** [comment]
After further testing, it seems that the issue is specifically triggered by using a non-default `TextAlignment` property at the same time as a non-default `HorizontalAlignment` property **only with a custom font**. 

The issue will still occur if `HorizontalAlignment` is set on a parent control, i.e. below will still trigger this issue ->

```
    <Grid HorizontalAlignment="Center">
        <TextBlock TextAlignment="Center" VerticalAlignment="Center" FontSize="70" Foreground="#AFC0CB" Text="Test" ClipToBounds="False" />
    </Grid>
```

**Comment by Gillibald** [comment]
Fixed by: #17402

**Comment by Niaxor** [comment]
It sure is! Thanks :)
